peak_finder/driver.py

```diff
@@ -243,85 +277,83 @@ def run(self):  # pylint: disable=R0912, R0914, too-many-statements
-                            peaks.append(inds_map[anom.peak])
+
+                centers.append(line_anomaly.centers)
+                amplitudes.append(line_anomaly.amplitudes)
+                starts.append(line_anomaly.starts)
+                ends.append(line_anomaly.ends)
+                line_ids.append(
+                    np.ones(len(line_anomaly.centers)) * line_anomaly.line_id
+                )
+                channel_groups.append(line_anomaly.group_ids)
+
+                if self.params.structural_markers:
+                    for line_group in line_anomaly.anomalies:
+                        for group in line_group.groups:
+                            locs = np.vstack(
+                                [
+                                    getattr(group.position, f"{k}_locations")
+                                    for k in "xyz"
+                                ]
+                            ).T
+                            inds_map = group.position.map_locations
+
+                            for anom in group.anomalies:
+                                anom_locs.append(locs[inds_map[anom.peak]])
+                                inflect_down.append(inds_map[anom.inflect_down])
+                                inflect_up.append(inds_map[anom.inflect_up])
+                                anom_start.append(inds_map[anom.start])
+                                anom_end.append(inds_map[anom.end])
+                                peaks.append(inds_map[anom.peak])
 
             print("Exporting . . .")
             group_points = None
-            if group_center:
-                # Create reference values and color_map
-                group_map, color_map = {0: "Unknown"}, [[0, 0, 0, 0, 0]]
-                for ind, (name, group) in enumerate(channel_groups.items()):
-                    group_map[ind + 1] = name
-                    color_map += [[ind + 1] + hex_to_rgb(group["color"]) + [0]]
-
+            if centers:
                 group_points = Points.create(
                     self.params.geoh5,
                     name="Anomaly Groups",
-                    vertices=np.vstack(group_center),
+                    vertices=np.vstack(centers),
                     parent=output_group,
                 )
 
                 group_points.entity_type.name = self.params.ga_group_name
                 group_points.add_data(
                     {
-                        "amplitude": {"values": np.asarray(amplitude)},
-                        "start": {
-                            "values": np.vstack(group_start).flatten().astype(np.int32)
-                        },
-                        "end": {
-                            "values": np.vstack(group_end).flatten().astype(np.int32)
-                        },
+                        "amplitude": {"values": np.hstack(amplitudes)},
+                        "start": {"values": np.hstack(starts).astype(np.int32)},
```

Comment:
Not really related to this PR, but can we support int64 in geoh5py and just do the conversion under the hood?

Reply:
Comes in as float right now, so need to convert regardless.